app/src/main/java/com/karumi/ui/presenter/SuperHeroesPresenter.kt

```diff
@@ -1,22 +1,42 @@
 package com.karumi.ui.presenter
 
+import co.metalab.asyncawait.async
 import com.karumi.ui.LifecycleSubscriber
+import com.karumi.ui.domain.Result.Left
+import com.karumi.ui.domain.Result.Right
+import com.karumi.ui.domain.model.SuperHero
+import com.karumi.ui.domain.usecase.GetSuperHeroes
 import java.lang.ref.WeakReference
 
-class SuperHeroesPresenter(view: View) : LifecycleSubscriber {
+class SuperHeroesPresenter(view: View, getSuperHeroes: GetSuperHeroes) : LifecycleSubscriber {
     private val viewWeak: WeakReference<View> = WeakReference(view)
+    private val getSuperHeroes: GetSuperHeroes = getSuperHeroes
 
     private fun view() = viewWeak.get()
 
     override fun initialize() {
-        view()?.hideLoading()
     }
 
     override fun update() {
+        refreshSuperHeroes()
+    }
 
+    private fun refreshSuperHeroes() {
+        async {
```

Comment:
The usage of async/await is so cool and so fancy that I love it ❤️  But I was wondering if we can easily configure an idling resource. I've been reading [the documentation](https://github.com/metalabdesign/AsyncAwait) and I don't see any reference to this topic. If this is the case, we will need to create our own idling resource. For this kata this approach is ok, but w/o the idling resource I'd not use this library.

Reply:
For the kata is not mandatory, and I like try wrap the async/await in an object. Let me check one idea that I have :P

Reply:
For the kata is not mandatory. That's why I don't request changes and just added comments 😃  I've created a new entry in our outreach ideas log for implementing this.